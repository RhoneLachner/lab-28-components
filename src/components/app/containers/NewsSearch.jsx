import React, { Component } from 'react';
import Search from '../Search';
import Article from '../Article';
import ArticleList from '../ArticleList';

export default class NewsSearch extends Component {
  render() {
    return (
      <>

        <div>
          <Search/>
        </div>

        <div>
          <Article />
        </div>

        <div>
          <ArticleList />
        </div>

      </>
    );
  }
}
