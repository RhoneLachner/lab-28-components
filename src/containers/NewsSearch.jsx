import React, { Component } from 'react';
import Search from '../components/Search';
import ArticleList from '../components/articles/ArticleList';
import { getArticleBySearch } from '../services/ArticleApi';

export default class NewsSearch extends Component {

    state = {
      search: '',
      articles: []
    }
    handleChange = ({ target }) => {
      this.setState({ search:target.value });
      getArticleBySearch(target.value)
        .then(articles => this.setState({ articles }));
    }
 
    render() {
      const { search, articles } = this.state;
      return (
        <>
          <div>
            <Search
              search={search}
              onChange={this.handleChange}
            />
          </div>

          <div>
            <ArticleList 
              articles={articles}
            />
          </div>

        </>
      );
    }
}
