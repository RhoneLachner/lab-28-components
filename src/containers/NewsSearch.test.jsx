import React from 'react';
import { render, cleanup } from '@testing-library/react';
import NewsSearch from './NewsSearch';

describe('NewsSearch container', () => {
  afterEach(() => cleanup());
  it('renders NewsSearch', () => {
    const { asFragment } = render(<NewsSearch />);
    expect(asFragment()).toMatchSnapshot();
  });
});
