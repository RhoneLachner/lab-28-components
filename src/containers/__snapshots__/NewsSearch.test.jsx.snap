// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsSearch container renders NewsSearch 1`] = `
<DocumentFragment>
  <div>
    <input
      type="text"
      value=""
    />
  </div>
  <div>
    <ul />
  </div>
</DocumentFragment>
`;
